### PR TITLE
Modify pixhack v3 to pixhack v3x

### PR DESCRIPTION
**Pull request by cuhome**
We have discontinued the pixhack v3, and now sell pixhack v3x. It is different from the led driver chip used by pixhack v3, and hmc5983 is also canceled. No other changes.

### CODE REVIEW

File: en/SUMMARY.md [hunk 1]
```diff
@@ -144,7 +145,7 @@
     * [Pixracer](flight_controller/pixracer.md)
     * [Cube (Pixhawk 2)](flight_controller/pixhawk-2.md)
     * [Pixhawk 3 Pro](flight_controller/pixhawk3_pro.md)
-    * [Pixhack v3](flight_controller/pixhack_v3.md)
+    * [Pixhack v3x](flight_controller/pixhack_v3x.md)
```
Comment (hamishwillee) [inline en/SUMMARY.md]: You probably could have kept the file name the same.

File: en/assembly/quick_start_pixhack_v3x.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+#Pixhack V2/V3/V3X wiring quick start
```
Comment (hamishwillee) [inline en/assembly/quick_start_pixhack_v3x.md]: Consistent headings. This should be first-letter of every major word capitalised. 

> Pixhack V2/V3/V3X Wiring Quick Start

File: en/assembly/quick_start_pixhack_v3x.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+The quick start guide shows how to power the Pixhack V2/V3 flight controller and connect its most important peripherals.
+
+
+![Pixhack v3x](../../assets/flight_controller/pixhack_v3x/pixhack_v3x.png)
+
+###Mount and Orient Controller
```
Comment (hamishwillee) [inline en/assembly/quick_start_pixhack_v3x.md]: Not critical, but good to have a space after #

File: en/assembly/quick_start_pixhack_v3x.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+Pixhack V2/V3/V3X should be mounted on the frame using vibration-damping foam pads (included in the kit). It should be positioned as close to your vehicle’s center of gravity as possible, oriented top-side up with the arrow pointing towards the front of the vehicle.
+
+<img src="../../assets/flight_controller/pixhack_v3x/pixhack_v3_vehicle_feont.jpg" align="center"/>
```
Comment (hamishwillee) [inline en/assembly/quick_start_pixhack_v3x.md]: File name appears to have typo error: _feont.jpg (ie you probably mean "front"). It does match file.

File: en/assembly/quick_start_pixhack_v3x.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+> **Note** If the controller cannot be mounted in the
+  recommended/default orientation (e.g. due to space constraints) you will
+  need to configure the autopilot software with the orientation that you
+  actually used: [Flight Controller Orientation](../config/flight_controller_orientation.md).
+  
+###Wiring Overview
+
+The image below shows how to connect the most important sensors and peripherals. We'll go through each of these in detail in the following sections.
+
+<img src="../../assets/flight_controller/pixhack_v3x/pixhack_v3_wiring_overvie.jpg" align="center"/>
```
Comment (hamishwillee) [inline en/assembly/quick_start_pixhack_v3x.md]: File name appears to have typo error: _overvie.jpg (ie you probably mean "overview"). It does match file.

File: en/assembly/quick_start_pixhack_v3x.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+###GPS + Compass
+
+The GPS/Compass should be connected to the pixhack v3  GPS interface and placed as far away from other electronic devices as possible on the frame, with the direction marker facing the front of the vehicle (separating the compass from other electronic devices will reduce interference).
+
+![Connect compass/GPS to Pixhack v3](../../assets/flight_controller/pixhack_v3x/pixhack_v3_gps.jpg) 
+
+###Safety Switch and power
+
+The cuav safety switch is an integrated safety switch and buzzer and should be connected to the pixhack v3 safe interface;Power module should connect to pixhack v3 power interface.
```
Comment (hamishwillee) [inline en/assembly/quick_start_pixhack_v3x.md]: cuav -> CUAV or even better, *CUAV safety switch*.

File: en/assembly/quick_start_pixhack_v3x.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+![Connect bazzer and safety switch to Pixhack v3](../../assets/flight_controller/pixhack_v3x/pixhack_v3_power_and_safe.jpg)
```
Comment (hamishwillee) [inline en/assembly/quick_start_pixhack_v3x.md]: buzzer, not bazzer

**Comment by hamishwillee** [comment]
A few typographic errors, but on the face of it good. 

@davids5 @jinger26  Is PX4 port stable enough with this hardware to warrant inclusion?

**Comment by cuhome** [comment]
error
rnroo
There is a mistake, I don't know how to deal with it, can you help me?

**Comment by cuhome** [comment]
@hamishwillee

**Comment by davids5** [comment]
> A few typographic errors, but on the face of it good.
> 
> @davids5 @jinger26 Is PX4 port stable enough with this hardware to warrant inclusion?

@hamishwillee, @cuhome  what are the differences from a pixhack v3 to a v3x?

I had added https://github.com/PX4/Firmware/pull/10231, to discriminate and ID the pixhack 3. I believe the rest of the issues (rotations) were resolved. But best ask @dagar

**Comment by cuhome** [comment]
@davids5 It just replaced the led driver chip (use ncp5623c) and removed the hmc5983.

**Comment by cuhome** [comment]
@davids5  Can you answer me what is the meaning of this error tone?
vedio:https://youtu.be/yEHutZGIvN8
The firmware is 1.90master and this alarm does not exist in 1.82stable and old firmware.

**Comment by davids5** [comment]
What you are hearing is [PX4_DEFINE_TUNE(16, SD_INIT, "MBAGP",  false /*  Make FS */)](https://github.com/PX4/Firmware/blob/master/src/lib/tunes/tune_definition.desc#L104)

There is an issue with theSD card.

The tunes are listed [here](https://github.com/PX4/Firmware/blob/master/src/lib/tunes/tune_definition.desc#L90)

You can  see the tune that played on the console.


FMUv2 ver 0x8 : Rev 0 V2
[boot] Fault Log info File No 4 Length 3177 flags:0x01 state:1
[boot] Fault Log is Armed
sercon: Registering CDC/ACM serial driver
sercon: Successfully registered the CDC/ACM serial driver
HW arch: PX4_FMU_V3
HW type: V2
HW version: 0x00090008
HW revision: 0x00000000
FW git-hash: 80fbf127021c4decf4e57ff464d690b667cc1294
FW version: 1.9.0 80 (17367168)
FW git-branch: master_nuttx_7.28+_rt
OS: NuttX
OS version: Release 7.28.0 (119275775)
OS git-hash: 1e6b485e4afbdcfe4353c203f49e751e6e8cb196
Build datetime: Jan 30 2019 01:42:53
Build uri: localhost
Toolchain: GNU GCC, 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
PX4GUID: 000100000000303533333036510400410026
MCU: STM32F42x, rev. 3
nsh: mount: mount failed: No such device
**INFO  [tune_control] Publishing standard tune 16
nsh: mkfatfs: mkfatfs failed: No such device**
INFO  [tune_control] Publishing standard tune 2
INFO  [param] selected parameter default file /fs/mtd_params
Board defaults: /etc/init.d/rc.board_defaults
WARN  [dataman] Could not open data manager file /fs/microsd/dataman
ERROR [dataman] dataman start failed
rgbled on I2C bus 2 at 0x55 (bus: 100 KHz, max: 100 KHz)
Board sensors: /etc/init.d/rc.board_sensors
HMC5883_I2C on I2C bus 1 at 0x1e (bus: 100 KHz, max: 400 KHz)
WARN  [lis3mdl] no device on bus 2
INFO  [ist8310] no device on bus 1
INFO  [ist8310] no device on bus 5
WARN  [hmc5883] no device on bus 2 (type: 1)
WARN  [mpu6000] no device on bus #3 (SPI1)
MPU6000 on SPI bus 1 at 4 (1000 KHz)
WARN  [mpu6000] no device on bus #5 (SPI4)
INFO  [mpu9250] Bus probed: 2
WARN  [mpu9250] no device on bus 2
INFO  [mpu9250] Bus probed: 3
WARN  [mpu9250] probe failed! 104
WARN  [mpu9250] no device on bus 3
INFO  [mpu9250] Bus probed: 5
WARN  [mpu9250] probe failed! 255
WARN  [mpu9250] no device on bus 5
L3GD20 on SPI bus 1 at 1 (11000 KHz)
LSM303D on SPI bus 1 at 2 (11000 KHz)
MS5611_SPI on SPI bus 4 at 3 (20000 KHz)
WARN  [ms5611] no device on bus 4
MS5611_SPI on SPI bus 1 at 3 (20000 KHz)
WARN  [bst] no devices found
INFO  [mpu9250] Bus probed: 2
WARN  [mpu9250] no device on bus 2
px4flow [138:100]
INFO  [px4flow] scanning I2C buses for device..
INFO  [mavlink] mode: Config, data rate: 800000 B/s on /dev/ttyACM0 @ 57600B
ERROR [mavlink] DM_KEY_MISSION_STATE lock failed
ERROR [mavlink] offboard mission init failed (-1)
Starting Main GPS on /dev/ttyS3
Starting MAVLink on /dev/ttyS1
INFO  [mavlink] mode: Normal, data rate: 1200 B/s on /dev/ttyS1 @ 57600B
INFO  [px4io] default PWM output device
INFO  [init] Mixer: /etc/mixers/quad_x.main.mix on /dev/pwm_output0
INFO  [init] Mixer: /etc/mixers/pass.aux.mix on /dev/pwm_output1
INFO  [logger] logger started (mode=all)
INFO  [logger] log root dir created: /fs/microsd/log

NuttShell (NSH)
nsh> INFO  [ecl/EKF] EKF aligned, (pressure height, IMU buf: 22, OBS buf: 14)

on the console you can type `tune_control play -t [1-20] and hear each one.
i.e. The error tune 'tune_control play -t 2`


What you are hearing is PX4_DEFINE_TUNE(16, SD_INIT, "MBAGP",  false /*  Make FS */)(https://github.com/PX4/Firmware/blob/master/src/lib/tunes/tune_definition.desc#L104)

**Comment by davids5** [comment]
@hamishwillee is the above documented anywhere? I think it should be. With sound clips :)

**Comment by hamishwillee** [comment]
> @hamishwillee, @cuhome what are the differences from a pixhack v3 to a v3x?

@davids5 Leaving this one for @cuhome - I only know what is in this doc.

> I had added PX4/Firmware#10231, to discriminate and ID the pixhack 3. I believe the rest of the issues (rotations) were resolved. But best ask @dagar

@dagar Can you please confirm that this board is stable enough on PX4 to include in docs? 

> is the above documented anywhere? I think it should be. With sound clips :)

@davids5 No, looked at this extensively yesterday and responded similarly to you on slack (ie with link to tunes). I couldn't find a player, and simulator does not pay the tunes so gave up. Added https://github.com/PX4/px4_user_guide/issues/426 and I will try add something in the coming days.

**Comment by davids5** [comment]
> @davids5 No, looked at this extensively yesterday and responded similarly to you on slack (ie with link to tunes). I couldn't find a player, and simulator does not pay the tunes so gave up. Added #426 and I will try add something in the coming days.

@hamishwillee - Sorry I did not see anything on slack. Do you have HW? If so you can use it as a player
type `tune_control stop;tune_control play -t n` where n is 1:20 on the console (or mavlink console)

**Comment by hamishwillee** [comment]
> @hamishwillee - Sorry I did not see anything on slack. Do you have HW? If so you can use it as a player
type tune_control stop;tune_control play -t n where n is 1:20 on the console (or mavlink console)

@davids5 - yes, I have hardware. Just a bit short on time after taking a week off. Should be able to try this next week. Thanks for your help.

File: en/SUMMARY.md [hunk 8]
```diff
@@ -13,6 +13,7 @@
   * [Mounting the Flight Controller](assembly/mount_and_orient_controller.md)
   * [Pixhawk 4 Wiring Quickstart](assembly/quick_start_pixhawk4.md)
   * [Pixhawk 4 Mini Wiring Quickstart](assembly/quick_start_pixhawk4_mini.md)
+  * [Pixhack v3 Wiring Quickstart](assembly/quick_start_pixhack_v3x.md)
```
Comment (dagar) [inline en/SUMMARY.md]: v3x?

File: en/flight_controller/pixhack_v3.md [hunk 9]
```diff
@@ -15,9 +18,9 @@ The board is a variant of the SOLO Pixhawk<sup>&reg;</sup> 2 (PH2) flight contro
   * STM32F427
   * STM32F100 (Failsafe co-processor)
 * Sensors:
-  * Accelerometers (3): LS303D, MPU6000, MPU9250/hmc5983
-  * Gyroscopes (3): L3GD20, MPU6000, MPU9250
-  * Compass (2): LS303D, MPU9250
+  * Accelerometers (2): LS303D, MPU6000
```
Comment (dagar) [inline en/flight_controller/pixhack_v3.md]: Isn't it an mpu9250?

**Comment by hamishwillee** [comment]
@davids5 FYI, tunes 1-15 work fine, but tunes 16-20 do not play using the `-t` option (using Pixhawk 4 mini with buzzer, setting tune via QGC MAVlink console). The tunes do play with the -m option, so this does not affect me, but does indicate a bug.

**Comment by davids5** [comment]
@hamishwillee did you uses `tune_control stop` first before every command? 

It works on FMUv2 and resent master (on my rt branch)

```
nsh> ver all
HW arch: PX4_FMU_V3
HW type: V2
HW version: 0x00090008
HW revision: 0x00000000
FW git-hash: 37c8dc9edc55c38f729cf5fa9e3c4db779fe1e73
FW version: 1.9.0 80 (17367168)
FW git-branch: master_nuttx_7.28+_rt
OS: NuttX
OS version: Release 7.28.0 (119275775)
OS git-hash: 2f18326c0936000b1d410ad5fc6ebe1e1d0de671
Build datetime: Feb  2 2019 04:35:42
Build uri: localhost
Toolchain: GNU GCC, 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
PX4GUID: 000100000000303533333036510400410026
MCU: STM32F42x, rev. 3

nsh>
nsh> tune_control stop;tune_control play -t 16
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 16
nsh> tune_control stop;tune_control play -t 17
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 17
nsh> tune_control stop;tune_control play -t 18
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 18
nsh> tune_control stop;tune_control play -t 19
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 19
nsh>  tune_control stop;tune_control play -t 16
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 16
nsh>  tune_control stop;tune_control play -t 17
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 17
nsh>  tune_control stop;tune_control play -t 18
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 18
nsh>  tune_control stop;tune_control play -t 19
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 19
nsh>  tune_control stop;tune_control play -t 20
INFO  [tune_control] Stopping playback...
INFO  [tune_control] Publishing standard tune 20
nsh>  tune_control stop;tune_control play -t 21
INFO  [tune_control] Stopping playback...
Usage: tune_control <command> [arguments...]
 Commands:

   play          Play system tune, tone, or melody
     [-t <val>]  Play predefined system tune
                 default: 1
     [-f <val>]  Frequency of tone in Hz (0-22kHz)
                 default: 0
     [-d <val>]  Duration of tone in us
                 default: 1
     [-s <val>]  Strength of tone (0-100)
                 default: 40
     [-m <val>]  Melody in string form
                 values: <string> - e.g. "MFT200e8a8a"

   libtest       Test library

   stop          Stop playback (use for repeated tunes)

```

File: en/flight_controller/pixhack_v3.md (hunk 9 above)
Comment (davids5) [reply en/flight_controller/pixhack_v3.md]: @cuhome - Please list the devices on the pixhack v3 and v3x so it may be documented correctly.

**Comment by davids5** [comment]
@hamishwillee 
> FYI only, if you play a tune without calling stop, the first tune is stopped anyway.

Is that true if `interruptible` in the table is set false?

**Comment by hamishwillee** [comment]
>> FYI only, if you play a tune without calling stop, the first tune is stopped anyway.

> Is that true if `interruptible` in the table is set false?

No idea. I just tested on the fmuv5 firmware for master. This is not something I documented because it isn't of interest to end users.

File: en/flight_controller/pixhack_v3.md (hunk 9 above)
Comment (hamishwillee) [reply en/flight_controller/pixhack_v3.md]: @cuhome I guess you're away on Chinese new year. This will wait for your return.

**Comment by cuhome** [comment]
@hamishwillee @dagar @davids5 Both pixhack v3x and pixhack v3 don't have mpu9250. It uses two mpu6000s, of which pixhack v3 has hmc5983 built in, and pixhack v3x has no hmc5983.

**Comment by davids5** [comment]
@cuhome - thank you 

@dagar do we have this covered correctly? See https://github.com/PX4/Firmware/commit/c6a502f150c9f3865a453f131cbc6fb35c82384c#r32304242

**Comment by hamishwillee** [comment]
> @dagar do we have this covered correctly? See [PX4/Firmware@c6a502f#r32304242](https://github.com/PX4/Firmware/commit/c6a502f150c9f3865a453f131cbc6fb35c82384c#r32304242)

And if so, please confirm that you are good to merge this PR.

**Comment by cuhome** [comment]
@davids5 @dagar Oh, no. Have you modified the 1.82 stble firmware? I tested it, pixhack v3x is considered v2m, does not recognize lsm303d and ist8310.

**Comment by cuhome** [comment]
@dagar @hamishwillee @davids5 This is the debug information I used to load different firmware using pixhack v3x. Why is hw arch different? What will this difference lead to? I tested and found that using the latest 1.90master firmware, my flight controller will indicate that the acceleration is inconsistent .If you use 1.82 stable firmware, this error message will not appear, but it cannot drive ist8310.
ddd

ggg
@hamishwillee

**Comment by davids5** [comment]
@cuhome 

> This is the debug information I used to load different firmware using pixhack v3x. Why is hw arch different? 

In 1.82 there was one build for px4_fmu-v2 and px4_fmu-v3. It produced 2 FW files  px4_fmuv2_defualt.px4 and px4_fmu-v3_defualt. Both ID as PX4_FMU_V2. 
The v3 .px4 was for HW that can support a 2MB of FLASH The v2 px4 file was for old STM32 with the FLASH errata. Where a max of 1 MB could be used.

In 1.9 there are 2 builds. That ID as PX4_FMU_V2  and PX4_FMU_V3. 

The proper build for the PH V3x is the the px4_fmu-v3 build. It will ID as PX4_FMU_V3 the board should be detected as HW Type: V30 and it was.

What will this difference lead to? 

The delta from 1.82 it to large for me to enumerate. But the OS is updates and more drivers/features were added and for the PH3 HW the ist8310 is started.

>I tested and found that using the latest 1.90master firmware, my flight controller will indicate that the acceleration is inconsistent 

Please provide details

> If you use 1.82 stable firmware, this error message will not appear, but it cannot drive ist8310.

That makes sense as the driver was not started in 1.82 for that HW.

Please target 1.9 for your testing and report your findings.

**Comment by cuhome** [comment]
@davids5 davids5What settings can I use to see three sets of accelerometer data in real time? I can only look at a set of sensor data on qgc and can't find the difference.

**Comment by davids5** [comment]
> @davids5 davids5What settings can I use to see three sets of accelerometer data in real time? I can only look at a set of sensor data on qgc and can't find the difference.

@dagar - would you please lend a hand here.

**Pull request closed without merging**